tools/get-keys-from-ts.js

```diff
@@ -0,0 +1,546 @@
+
+        /**
+         * Alphabetize
+         * @param {string} typeA The first type to compare
+         * @param {string} typeB The second type to compare
+         * @returns {1|-1} The sorting index
+         */
+        return function sortKeys(typeA, typeB) {
+            if (!sequence) {
+                return typeA < typeB ? -1 : 1;
+            }
+
+            const idxA = sequence.indexOf(typeA);
+            const idxB = sequence.indexOf(typeB);
+
+            if (idxA === -1 && idxB === -1) {
+                return typeA < typeB ? -1 : 1;
+            }
+            if (idxA === -1) {
+                return 1;
+            }
+            if (idxB === -1) {
+                return -1;
+            }
+
+            return idxA < idxB ? -1 : 1;
+        };
+    }
+
+    for (const [type, keys] of sortedNodeEntries) {
+        keys.sort(getKeySorter(type));
+    }
+
+    return Object.fromEntries(sortedNodeEntries);
+}
+
+/**
+ * Traverse interface `extends`
+ * @param {Node} declNode The TS declaration node
+ * @param {Function} handler The callback
+ * @returns {any[]} Return value of handler
+ */
+function traverseExtends(declNode, handler) {
+    const ret = [];
+
+    for (const extension of declNode.extends || []) {
+        const { typeParameters, expression } = extension;
+        const innerInterfaceName = expression.name;
+
+        let res;
+
+        if (typeParameters) {
+            if (innerInterfaceName !== "Omit") {
+                throw new Error("Unknown type parameter");
+            }
+
+            const [param, ...excludedAST] = typeParameters.params;
+            const paramInterfaceName = param.typeName.name;
+            const excluded = excludedAST.flatMap(findOmitTypes);
+
+            res = handler({ iName: paramInterfaceName, excluded });
+        } else {
+            res = handler({ iName: innerInterfaceName });
+        }
+
+        ret.push(res);
+    }
+
+    return ret;
+}
+
+/**
+ * Traverse the properties of a declaration node.
+ * @param {Node} tsDeclarationNode The declaration node
+ * @param {(string) => void} handler Passed the property
+ * @returns {any[]} The return values of the callback
+ */
+function traverseProperties(tsDeclarationNode, handler) {
+    const tsPropertySignatures = tsDeclarationNode.body.body;
+
+    const ret = [];
+
+    for (const tsPropertySignature of tsPropertySignatures) {
+        const property = tsPropertySignature.key.name;
+
+        const tsAnnotation = tsPropertySignature.typeAnnotation.typeAnnotation;
+
+        const res = handler({ property, tsAnnotation });
+
+        ret.push(res);
+    }
+
+    return ret;
+}
+
+/**
+ * Builds visitor keys based on TypeScript declaration.
+ * @param {string} code TypeScript declaration file as code to parse.
+ * @param {{supplementaryDeclarations: Node[]}} [options] The options
+ * @returns {Promise<VisitorKeysExport>} The built visitor keys
+ */
+async function getKeysFromTs(code, {
```

Comment:
Does this function need to be `async`?

Reply:
Actually, no. Removed and changed the return type. `require-await` would be a good rule to be adding to `eslint-config-eslint` too I think.